src/Acuminator/Acuminator.Utilities/DiagnosticSuppression/SuppressionManager.cs

```diff
@@ -204,14 +209,61 @@ private static SuppressionDiffResult CompareFiles(SuppressionFile oldFile, Suppr
 		{
 			cancellation.ThrowIfCancellationRequested();
 
-			if (settings.SuppressionMechanismEnabled && Instance?.IsSuppressed(semanticModel, diagnostic, cancellation) == true)
+			if (settings.SuppressionMechanismEnabled && 
+			    (Instance?.IsSuppressed(semanticModel, diagnostic, cancellation) == true ||
+			    CheckSuppressedComment(diagnostic, cancellation)))
 			{
 				return;
 			}
 
 			reportDiagnostic(diagnostic);
 		}
 
+		private static bool CheckSuppressedComment(Diagnostic diagnostic, CancellationToken cancellation)
+		{
+			cancellation.ThrowIfCancellationRequested();
+			SyntaxNode root = diagnostic.Location.SourceTree.GetRoot();
+			SyntaxNode node;
+
+			bool containsComment = false;
+
+			// Climb to the hill. Looking for comment on parents nodes.
+			for (node = root.FindNode(diagnostic.Location.SourceSpan); node != root ; node = node.Parent)
+			{
+				containsComment = CheckSuppressionCommentOnNode(diagnostic, node, cancellation);
+				
+				if (node is StatementSyntax || node is MemberDeclarationSyntax || containsComment)
+					break;
+			}
+
+			return containsComment;
+		}
+
+		private static bool CheckSuppressionCommentOnNode(Diagnostic diagnostic, SyntaxNode node, CancellationToken cancellation)
+		{
+			cancellation.ThrowIfCancellationRequested();
+
+			if (node != null && node.HasLeadingTrivia && node.GetLeadingTrivia().Any(SyntaxKind.SingleLineCommentTrivia))
```

Comment:
HasLeadingTrivia internally also call GetLeadingTrivia. It is actually implemented this way:
```csharp
node.GetLeadingTrivia().Count > 0;
```
Also you make yet another call to GetLeadingTrivia below. Isn't it possible to solve with one call:
```csharp
private static bool CheckSuppressionCommentOnNode(Diagnostic diagnostic, string diagnosticShortName, SyntaxNode node, CancellationToken cancellation)
{
	cancellation.ThrowIfCancellationRequested();
	var successfulMatch = node?.GetLeadingTrivia()
		.Where(x => x.RawKind == (int)SyntaxKind.SingleLineCommentTrivia)
		.Select(trivia => _suppressPattern.Match(trivia.ToString()))
		.FirstOrDefault(match => match.Success &&
		    diagnostic.Id == match.Groups[1].Value && diagnosticShortName == match.Groups[2].Value);
	return successfulMatch != null;
}
```